pulp_smash/tests/rpm/api_v2/test_rsync_distributor.py

```diff
@@ -0,0 +1,228 @@
+# coding utf-8
+"""Test the functionality of the RPM rsync distributor.
+
+The RPM rsync distributor lets one publish content units to a directory via
+rsync+ssh. The RPM rsync distributor can be used by itself, but the Pulp user
+story and configuration parameters make it clear that the primary use case is
+as a post-distributor. In other words, it's possible to attach both a yum
+distributor and rsync distributor to a repository, and to use only the latter
+for publishes, but most of the interesting use cases arise when both are used
+for publishes.
+
+For more information on the RPM rsync distributor, see `Pulp #1759`_.
+
+.. _Pulp #1759: https://pulp.plan.io/issues/1759
+"""
+from __future__ import unicode_literals
+
+from pulp_smash import api, cli, utils
+from pulp_smash.compat import urljoin, urlparse
+from pulp_smash.constants import REPOSITORY_PATH, RPM_FEED_URL
+from pulp_smash.tests.rpm.api_v2.utils import gen_distributor, gen_repo
+from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+
+def _make_user(cfg):
+    """A generator to create a user account on the target system.
+
+    Yield a username and private key. The corresponding public key is make the
+    one and only key in the user's ``authorized_keys`` file.
+
+    The username and private key are yielded one-by-one rather than as a pair
+    because the user creation and key creation steps are executed serially.
+    Should the latter fail, the calling function will still be able to delete
+    the created user.
+
+    The user is given a home directory. When deleting this user, make sure to
+    pass ``--remove`` to ``userdel``. Otherwise, the home directory will be
+    left in place.
+    """
+    client = cli.Client(cfg)
+    sudo = '' if utils.is_root(cfg) else 'sudo '
```

Comment:
I've seen this logic on every place SSH is used, we should think on a way to make it generic or an util function like `sudo = utils.sudo(cfg)` or maybe some argument to `SSHClient.run`, something like `client.run([...], sudo=True)`. Just leaving some thoughts for later improvement.